src/sql/magic_plot.py

```diff
@@ -73,23 +74,40 @@ def execute(self, line="", cell="", local_ns=None):
         column = util.sanitize_identifier(column)
         table = util.sanitize_identifier(cmd.args.table)
 
+        if cmd.args.with_:
+            warnings.warn(
+                "--with will be deprecated soon, and dependencies "
+                "will be inferred from saved queries instead",
+                FutureWarning,
+            )
```

Comment:
I suggested changing this warning message so let's move it to a single place and call it from there:

e.g., 
```python
        if cmd.args.with_:
            show_with_deprecation_warning()
```